src/DataStructures/DynamicBuffer.hpp

```diff
@@ -28,14 +28,19 @@ class er;
  * size of the vector is known at compile time, a `TempBuffer` object should be
  * used instead.
  *
- * If needed it should be fairly straightforward to generalize to
- * `ComplexDataVector`.
+ * Currently this can only be used for `DataVector`s, but if needed, it should
+ * be fairly straightforward to generalize to `ComplexDataVector`.
  */
 template <typename T>
 class DynamicBuffer {
  public:
+  // Rather than having to template DynamicBuffer on the static size, just use
+  // the default static size since this only determines if something is derived
+  // from VectorImpl and not the specific size itself
   static constexpr bool is_data_vector_type =
-      std::is_base_of_v<VectorImpl<tt::get_fundamental_type_t<T>, T>, T>;
+      std::is_base_of_v<VectorImpl<tt::get_fundamental_type_t<T>, T,
+                                   default_vector_impl_static_size>,
+                        T>;
```

Comment:
I don't understand.  Isn't this checking that the static size is the default?

Reply:
Yes. So currently, `DynamicBuffer` is only used in `YlmSpherepack`. And `DynamicBuffer` is only instantiated for `double, DataVector`. No other types. So we want to check if it's using the default because that's what `DataVector` uses. If in the future we want to extend `DynamicBuffer` to other vector types that don't use the default static size, then we can do that.

Reply:
If it's only supposed to work for DataVector just check if it's a DataVector instead of doing stuff with VectorImpl inheritance.